src/components/posts/PostList.js

```diff
@@ -1,22 +1,59 @@
 import PostItem from "./PostItem";
 import classes from "./PostList.module.css";
+import OrderPosts from "./OrderPosts.js";
+import { useState } from "react";
+
+function PostList({ posts }) {
+  const [orderUpDown, setOrderUpDown] = useState(true);
+  const [authorDate, setAuthorDate] = useState(true);
+
+  if (authorDate) {
+    orderByDate();
+    if (!orderUpDown) {
+      posts.reverse();
+    }
+  } else {
+    console.log(posts);
+    orderByName();
+    if (orderUpDown) {
+      posts.reverse();
+    }
+  }
+
+  function orderByDate() {
+    posts.sort(function (a, b) {
+      return new Date(b.date) - new Date(a.date);
+    });
+  }
+
+  function orderByName() {
+    posts.sort(function (a, b) {
+      return a.author.localeCompare(b.author);
+    });
```

Comment:
Needing to have all of the sorting information in this class feels like quite a large gap between where it is changed and setup compared with where it is defined. Can you think of a way to not need these conditional cases and move a lot of this logic into the `OrderPosts` class